packages/reactivity/__tests__/watch.spec.ts

```diff
@@ -260,4 +260,21 @@ describe('watch', () => {
     src.value = 10
     expect(spy).toHaveBeenCalledTimes(2)
   })
+
+  test('should ensure correct execution order in batch processing', () => {
+    const dummy: number[] = []
+    const n1 = ref(0)
+    const n2 = ref(0)
+    const sum = computed(() => n1.value + n2.value)
+    watch(n1, () => {
+      dummy.push(1)
+      n2.value++
+    })
```

Comment:
I don't understand this test. Without the `flush: 'sync'` I don't think it's testing the same scenario as the original bug report.

Reply:
I double-checked, and the root cause is the same in both cases. In the nested batch, the computed sub’s `next` property gets reset, causing the original `next` sub not to run.